checkov/dockerfile/checks/WorkdirIsAbsolute.py

```diff
@@ -0,0 +1,26 @@
+import re
+
+from checkov.common.models.enums import CheckCategories, CheckResult
+from checkov.dockerfile.base_dockerfile_check import BaseDockerfileCheck
+
+
+class WorkdirIsAbsolute(BaseDockerfileCheck):
+    def __init__(self):
+        """
+        For clarity and reliability, you should always use absolute paths for your WORKDIR.
+        """
+        name = "Ensure that WORKDIR values are absolute paths"
+        id = "CKV_DOCKER_10"
+        supported_instructions = ["WORKDIR"]
+        categories = [CheckCategories.CONVENTION]
+        super().__init__(name=name, id=id, categories=categories, supported_instructions=supported_instructions)
+
+    def scan_entity_conf(self, conf):
+        for mydir in conf:
+            mypath = mydir["value"]
+            if not re.match("(^/[A-z0-9-_+]*)|(^[A-z0-9-_+]:\\\\.*)|(^\\$[{}A-z0-9-_+].*)", mypath):
```

Comment:
please use `re.compile(...)` to create the regex as a "static" variable and then use it here, like here https://github.com/bridgecrewio/checkov/blob/f43ef11c3bc489cb1a28c24826ce4d778cf35813/checkov/arm/checks/resource/NSGRulePortAccessRestricted.py#L11

Reply:
have done it. Whats does that achieve?

Reply:
then the `flake8` plugin `dlint` is able to check, if we have some issue inside the regex